### ISSUE TITLE
Design tokens failing a11y automated CI checks

### ISSUE THREAD

**Issue by gd2910** [mi6/ic-ui-kit#3363]
## Summary 
Colour tokens have been [updated](https://github.com/mi6/ic-ui-kit/issues/2888) in line with Figma. This has caused some Cypress A11y checks to fail, mainly around some disabled states and destructive colours.

## 💬 Description 
Affected images;
Button
 - [x] destructive
 - [x] with-icon
 - [x] full-width
 - [x] focused-destructive
 - [x] dark-theme

Classification Banner
- [x] official
- [x] official-sensitive
- [x] secret
- [x] top-secret
- [x] country-official
- [x] up-to-official
- [x] inline-official

Footer
- [x] classification-banner

Status Tag
- [x] danger
- [x] small
- [x] large

Once the images or tests have been updated, uncomment the a11y checks

UPDATE: as part of this work, the Cypress tests for all components should be updated to ensure they use the same background in dark mode as is used in figma. This should be `#17191C` and is different to our `--ic-color-background-primary-dark`, which is a token used for some components that need a dark background color. Currently a range of different shades of black are used across the various component tests

**Comment by ad9242** [comment]
Some (not all) of these seem to be disabled variants. If the disabled ones are deemed to not need to meet the contrast criteria, suggest we move them into separate visual tests that do not include the accessibility check, but keep the check for all other variants

**Comment by mi6-577** [comment]
- Classification has been updated to introduce Theme property and have added 2 new component tokens for dark mode colours (specifically for grey and red) to ensure contrast ratios are adhered to.
- Documentation for Style:Colours has been updated ready for website update

This should address the Classification and Footer issues

**Comment by mi6-577** [comment]
Design will need a little more information to go on here.
If the background colour has been adjusted and the disabled variants have been excluded, what is left and do the tests have any further indication on exactly which variants do not comply please?

**Comment by GCHQ-Developer-847** [comment]
I've just had a look and it appears to be the destructive variants of the button and the danger variant of the status tag which are causing the checks to fail.

The tests say that the light destructive variant of the button and the light danger variant of the status tag only have a contrast ratio of 4.41:1 (#f4f4f5 against #d4351c).

And the dark destructive variant of the button has a contrast ratio of 3.39:1 (#f4f4f5 against #e54330).

**Comment by GCHQ-Developer-847** [comment]
Actually, I've looked at Figma now and I think the issues with the above component variants is that they don't match the colours on Figma.

The light destructive button variant and the light danger status tag variants should both have a text colour of `#ffffff` instead of `#f4f4f5`. And the dark destructive button variant should have `#0b0c0c` as its text colour instead of `#f4f4f5`. Looks like it's to do with the wrong semantic tokens being used.

So this is something which the developer who picks up this ticket will need to resolve - and I don't think we need anything more from Design on this ticket :)

**Comment by GCHQ-Developer-299** [comment]
@ad9242 has suggested this could be a 2-3 point ticket, and has started work on it. We'll still give a proper refinement with the team

**Comment by ad9242** [comment]
based on work done so far, i'd be closer to a 3

**Comment by mi6-577** [comment]
Have made numerous changes to ensure consistency;

Semantic token changes;
Border/neutral
	default-dark					arch-500
	hover-dark					arch-300
	pressed-dark					arch-200

Border/error
	default-dark					red-900
	hover-light					red-1000
	pressed-light					red-11000
	pressed-dark					red-600

Border/success
	hover-dark					green-800
	pressed-dark					green-700

Border/warning
	default-dark					yellow-800
	pressed-dark					yellow-600
	pressed-light					yellow-1200
	pressed-dark					yellow-500
